### PR TITLE
[menu] Add way to register menu items #106

### LINKED ISSUE #106: [menu] Add way to register menu items
Right now, to add menu items in the several openwisp modules, we require a lot of repetition of code:

https://github.com/openwisp/openwisp-users/blob/1e31ea5f4b3a92c1e3c4f783f89fd204a90f23a6/openwisp_users/apps.py#L13-L25

We should simplify this, allowing to register new items with a function, optionally allowing to specify the  position at which the items will be added in the menu list (append by default to the end, allow to insert after a specific app or model).

### PR DESCRIPTION
**Pull request by atb00ker**
Closes #106

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -314,6 +314,23 @@ Code utilities
 
 Generates an random string of 32 characters.
 
+``openwisp_utils.utils.add_default_menu_items``
```
Comment (nemesisdesign) [inline README.rst:317]: `add_default_menu_items` seems to not match the actual function name below

File: openwisp_utils/utils.py [hunk 2]
```diff
@@ -11,6 +11,14 @@ def get_random_key():
     return get_random_string(length=32)
 
 
+def add_menu_items(items, name_menu='OPENWISP_DEFAULT_ADMIN_MENU_ITEMS'):
```
Comment (nemesisdesign) [inline openwisp_utils/utils.py:14]: what about `register_menu_items`? This is also consistent with similar functions we added in the monitoring and notifications modules.

File: README.rst [hunk 3]
```diff
@@ -314,6 +314,23 @@ Code utilities
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Takes input of a list of models name and settings_menu name and adds
+them to the side navigation bar in the admin dashboard interface.
+
+Usage:
+
+.. code-block:: python
+
+    from openwisp_utils.utils import add_menu_items
```
Comment (nemesisdesign) [inline README.rst:327]: would be good to suggest to use this in the `ready()` method of an `AppConfig` class.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Thanks Ajay! See my comments below.

**Comment by atb00ker** [comment]
Updated this one! :smile:

**Review by nemesisdesign — APPROVED** [review]
Thanks @atb00ker :+1:!

**Pull request merged**